### PR TITLE
docs: Talk about the log option for the environment

### PR DESCRIPTION
**Pull request by maraisr**
Here I aim to tell the community about the logging option on Environment

### CODE REVIEW

File: docs/Modern-RelayEnvironment.md [hunk 1]
```diff
@@ -56,3 +56,37 @@ function handlerProvider(handle) {
   );
 }
 ```
+
+## Adding a `log` function
+
+You can get access to the inner life cycle stages of Relay, and log those. Super helpful for tracing, and debugging. For an in-depth look at what log events you can access, I urge you to try it out. Log a few things, and have a play. Alternively you can find a [list of possible events](https://github.com/facebook/relay/blob/37cee8039b888755e4ad4219e703be13a9a49ebb/packages/relay-runtime/store/RelayStoreTypes.js#L410) in the source code.
```
Comment (zth) [inline docs/Modern-RelayEnvironment.md:62]: I don't think the rest of the docs are in "I" form, so maybe drop that here too?

**Review by sibelius — APPROVED** [review]
this is great

I do miss the old log using console.groupCollapsed()

**Comment by maraisr** [comment]
Yeah! I mean I'd personally like a context concept to like be able to get variables or operation for the .next etc... because the only way to get that is to save it against the transactionID, but you can't WeakMap that. But I'll pr something for that in the future.

**Comment by tyao1** [comment]
Hi @maraisr! Thanks for adding the doc! 
Currently the log option is unstable and it was just changed a bit in https://github.com/facebook/relay/commit/95606c5ed. We will wait until the option is more stable before merging it into the docs.

**Pull request closed without merging**